pallets/gear/src/lib.rs

```diff
@@ -884,55 +891,55 @@ pub mod pallet {
                     actor,
-                    dispatch: queued_dispatch.into_incoming(gas_limit),
+                    precharged_dispatch,
                     origin: ProgramId::from_origin(source),
-                    gas_allowance: u64::MAX,
                     subsequent_execution,
                 };
 
-                let may_be_returned_context = (!subsequent_execution
-                    && actor_id == main_program_id)
-                    .then(|| MessageExecutionContext {
-                        subsequent_execution: true,
+                let subsequent_execution = !subsequent_execution && actor_id == main_program_id;
+                let may_be_returned_context =
+                    subsequent_execution.then(|| MessageExecutionContext {
+                        subsequent_execution,
                         ..message_execution_context.clone()
                     });
 
                 let journal =
                     match core_processor::prepare(&block_config, message_execution_context) {
-                        PrepareResult::Ok {
-                            context,
-                            pages_with_data,
-                        } => {
-                            #[cfg(feature = "lazy-pages")]
-                            let memory_pages = {
-                                let _ = pages_with_data;
-                                assert!(lazy_pages::try_to_enable_lazy_pages());
-                                Default::default()
-                            };
-                            #[cfg(not(feature = "lazy-pages"))]
-                            let memory_pages = match common::get_program_data_for_pages(
-                                actor_id.into_origin(),
-                                pages_with_data.iter(),
+                        PrepareResult::Ok(context) => {
+                            let memory_pages = match Self::get_and_track_memory_pages(
+                                &mut ext_manager,
+                                actor_id,
+                                &context.actor_data().pages_with_data,
                             ) {
-                                Ok(data) => data,
-                                Err(err) => {
-                                    log::error!(
-                                        "Page data in storage is in invalid state: {}",
-                                        err
-                                    );
-                                    continue;
-                                }
+                                None => continue,
+                                Some(m) => m,
                             };
 
-                            ext_manager.insert_program_id_loaded_pages(actor_id);
+                            let code = match Self::get_code(context.actor_data().code_id, actor_id)
+                            {
+                                None => continue,
```

Comment:
why `continue` everywhere? it's unreachable to don't have code for program

Reply:
Agree but we had earlier a record to log::error with corresponding message. After these changes logging happens inside `Self::get_code` so logic seems to be the same.

Reply:
I mean using `.unwrap_or_else(|| unreachable!())` as everywhere

Reply:
ok